File: map/exam/repository/Repository.java
package com.example.exam.repository;


    import com.example.exam.domain.Entity;

    import java.util.List;

    public interface Repository<ID, E extends Entity<ID>> {
        E findOne(ID id);
        List<E> findAll();
        E save(E entity);
        E delete(ID id);
        E update(E entity);
    }

File: mpp/Spectacole/repository/Repository.cs
using System.Collections.Generic;
using lab_mpp3.domain;

namespace lab_mpp3.repository
{
    public interface Repository<T, Tid> where T : Entity<Tid>
    {
        void save(T elem);
        void delete(T elem);
        void update(T elem, Tid id);
        T findOne(Tid id);
        IEnumerable<T> findAll(); // Specify the type parameter for IEnumerable
    }
}